src/torchmetrics/utilities/data.py

```diff
@@ -212,30 +212,32 @@ def apply_to_collection(
     return data
 
 
-def get_group_indexes(indexes: Tensor) -> List[Tensor]:
-    """Given an integer ``indexes``, return indexes for each different value in ``indexes``.
-
-    Args:
-        indexes:
-
-    Return:
-        A list of integer ``torch.Tensor``s
+def get_indexes_splits(indexes: torch.Tensor) -> List[int]:
```

Comment:
this function is just doing bin counting right?
then this function should be able to do it:
https://github.com/Lightning-AI/metrics/blob/a0943fd4480a338de8e5eb11af4f93feac67228a/src/torchmetrics/utilities/data.py#L249-L272

Reply:
I've edited the code to use a variation that calls `_bincount` but also allow to count on tensors that contain non contiguous values.